docs/user/adding-new-metrics.md

```diff
@@ -1,97 +1,216 @@
+Removing a metric does not affect the availability of data already collected by the pipeline.
+
+If the metric is still needed after its expiration date, it should go back for [another round of data review](https://wiki.mozilla.org/Firefox/Data_Collection) to have its expiration date extended.
+
+### When should Glean automatically clear the measurement?
+
+The `lifetime` parameter of a metric defines when its value will be cleared. There are three lifetime options available:
+
+- `ping` (default): The metric is cleared each time it is sent in the ping.
   This is the most common case, and should be used for metrics that are highly dynamic, such as things computed in response to the user's interaction with the application.
-- `application`: The metric is related to an application run, and is reset only when the application restarts.
+  
+- `application`: The metric is related to an application run, and is cleared every time the application restarts.
   This should be used for things that are constant during the run of an application, such as the operating system version.
   In practice, these metrics are generally set during application startup.
   A common mistake---using the `ping` lifetime for these type of metrics---means that they will only be included in the first ping sent during a particular run of the application.
+  
 - `user`: The metric is part of the user's profile.
   This should be used for things that change only when the user's profile is created.
   It is rare to use this lifetime outside of some metrics that are built in to Glean, such as `client_id`.
 
-### Naming things
+While lifetimes are important to understand for all metric types, they are particularly important for the metric types that record single values and don't aggregate on the client (`boolean`, `string`, `labeled_string`, `string_list`, `datetime` and `uuid`), since these metrics will send the "last known" value and missing the earlier values could be a form of unintended data loss.
+
+#### A lifetime example
+
+Let's work through an example to see how these lifetimes play out in practice. Let's suppose we have a user preference, "turbo mode", which defaults to `false`, but the user can turn it to `true` at any time.  We want to know when this flag is `true` so we can measure its affect on other metrics in the same ping.  In the following diagram, we look at a time period that sends 4 pings across two separate runs of the application. We assume here, that like Glean's built-in [metrics ping](pings/metrics.html), the developer writing the metric isn't in control of when the ping is submitted. 
+
+In this diagram, the ping "lifetimes" are represented as rectangles, but the moment the ping is "submitted" is represented by its right edge. The user changes the "turbo mode" setting from `false` to `true` at 3 seconds.
+  
+```mermaid
+gantt
+  axisFormat %S
+  title Metric lifetimes
+  dateFormat  s
+
+  section Metrics<br/>pings
+    Ping 1 :active, p1, 0, 2s
```

Comment:
Would it make sense to note the value reported by these metrics ping?

Reply:
Yes, but I'm not sure how to fit that in the visualization -- the value is different for each of the metric examples.  (I'm feeling a bit underwhelmed by mermaid for this -- I might need to fire up Inkscape and do it by hand, but I was trying to avoid that...)